tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1819,8 +1855,14 @@ static bool CombineBasicTypes(
       switch (uBits)
       {
       case BPROP_BITS8:  *pOutKind = AR_BASIC_UINT8;     break;
-      case BPROP_BITS16: (uEitherFlags & BPROP_MIN_PRECISION) ?
-        *pOutKind = AR_BASIC_MIN16UINT : *pOutKind = AR_BASIC_UINT16; break;
+      case BPROP_BITS16:
+        if (IS_BPROP_16BIT_AINT_NO_MIN_PRECISION(uLeftProps) ||
+          IS_BPROP_16BIT_AINT_NO_MIN_PRECISION(uRightProps)) {
```

Comment:
Why not just `BPROP_MIN_PRECISION & uBothFlags` (with reversed then/else)?  We already are under a path that guarantees both sides are integers.

Reply:
How about the scenario: `min16int + uint16_t` ?
I think the current change would return `uint16_t`, but your suggested change would return `min16uint` ?

Reply:
Discussed offline: the suggested change should produce the correct result.  It can be the original code with just `uEitherFlags` changed to `uBothFlags` in each of these cases.